### PR TITLE
Pulled in Payments Gateways, changed Federal Bank status, added Amazon link

### LINKED ISSUE #402: QR code to display in https://keralarescue.in/reg_contrib/
### Subject of the issue
CM relief fund redirection (https://donation.cmdrf.kerala.gov.in/) loading too slow and hence recommended to display the QR codes directly in this page,  https://keralarescue.in/reg_contrib/ for easy transfer

### PR DESCRIPTION
**Pull request by aravindharinair**
### Issue Reference
This PR addresses the Issue : Fixes #402 
This PR references Pull Request: #468 by @arjunkomath and adds his changes.

### Summarize

@arjunkomath has added payment links for all payment methods as per [CM's Disaster Relief Fund](https://donation.cmdrf.kerala.gov.in/)

I have edited to reflect Federal Bank's temporary outage and the addition of HDFC Bank to possible payment methods.

Also added Amazon's [Kerala Need Your Help](https://www.amazon.in/b?ie=UTF8&node=8891257031&pf_rd_p=2c1c1ac8-3c7f-4880-b5a7-e870ec58eefa&pf_rd_r=TEX7HJ9QSHSJQE3EX59W) reference link for material contributions.

> Verify bank details and payment methods
> Test on various devices

Note: Awaiting confirmation regarding Federal Bank's outage.

### CODE REVIEW

**Comment by riginoommen** [comment]
Please Squash the commits into one.

https://makandracards.com/makandra/527-squash-several-git-commits-into-a-single-commit

Right now the commit messages are duplicate and hard to track history

**Comment by aravindharinair** [comment]
@riginoommen will do.

**Comment by DextersHub** [comment]
Verified and Tested

**Comment by riginoommen** [comment]
@aravindharinair can you please reset the branch

**Comment by bobinson** [comment]
need one more person to test this please

**Comment by aravindharinair** [comment]
@riginoommen Success!!!

**Comment by riginoommen** [comment]
@aravindharinair https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

1) Reset the branch
2) switch to the resetted branch
3) if you want to make 1 commit from last 5 commits

First do `git log` and figure out the commits you need to sqash

then do `git rebase -i HEAD~5`

**Comment by drtechie** [comment]
image

**Comment by aravindharinair** [comment]
@riginoommen Is it ok now?

File: mainapp/templates/mainapp/contributor_form.html [hunk 1]
```diff
@@ -1,48 +1,199 @@
...
+                     Domestic Credit and Debit Cards supported<br/>
+                     <strong>INTERNATIONAL</strong> Credit and Debit Cards supported<br/><br/>
+
+                     payment charges: Nil
+                 </td>
+                 <td>
+                     A/c Number : 50100248534439<br/>
+                     A/c Name: Chief Minister’s Distress Relief Fund<br/>
+                     Branch: Vattiyoorkavu, Trivandrum<br/>
+                     IFSC : HDFC0002485</br>
```
Comment (bhumilharia) [inline mainapp/templates/mainapp/contributor_form.html]: `<br/>` ?

File: mainapp/templates/mainapp/contributor_form.html [hunk 2]
```diff
@@ -1,48 +1,199 @@
...
+                 <td>
+                     No QR Code
+                 </td>
+               </tr>
+    </tbody>
+  </table>
+  <br />
+  <h4>Donations in Kind</h4>
+  <p>
+      <img style="max-width:100%" src="https://images-eu.ssl-images-amazon.com/images/G/31/traffic/PC_editorial._CB1198675309_.jpg" alt="Amazon_Cover"/> </br></br>
```
Comment (bhumilharia) [inline mainapp/templates/mainapp/contributor_form.html]: `br` comment applicable in multiple places.

File: mainapp/templates/mainapp/contributor_form.html (hunk 1 above)
Comment (aravindharinair) [reply mainapp/templates/mainapp/contributor_form.html]: Both works. But will change it.

File: mainapp/templates/mainapp/contributor_form.html (hunk 2 above)
Comment (riginoommen) [reply mainapp/templates/mainapp/contributor_form.html]: @bhumilharia thanks for bringing it to notice. can we replace <p></p> with <br/>.

@aravindharinair

**Comment by bhumilharia** [comment]
I'm testing this now

**Comment by riginoommen** [comment]
@aravindharinair Commit history looks clean

File: mainapp/templates/mainapp/contributor_form.html (hunk 2 above)
Comment (aravindharinair) [reply mainapp/templates/mainapp/contributor_form.html]: trying to :D

**Comment by bhumilharia** [comment]
This is tested on Firefox (desktop) and Chrome (mobile and desktop).
I've verified the account numbers as well with the Kerala donation site.
All QR codes verified to be working (UPI QRs with PhonePe, and Paytm QR from Paytm app)

**Comment by aravindharinair** [comment]
@bhumilharia and @riginoommen changed <br /> but something else is wrong.

**Comment by riginoommen** [comment]
@aravindharinair what is the exact issue you face  currently

**Comment by aravindharinair** [comment]
The Travis build is failing.

**Comment by riginoommen** [comment]
@aravindharinair its probably because of the existing conflict

**Comment by DextersHub** [comment]
I think something is wrong here in the pull, your development branch was not synced with master branch before you made these changes I guess. Your diff shows modifications to 19 files.

**Comment by aravindharinair** [comment]
@DextersHub what about now?
Forgot to merge my branch with master

File: mainapp/templates/mainapp/contributor_form.html [hunk 3]
```diff
@@ -40,9 +37,161 @@
...
+            </td>
+            <td>
+                Airtel Payments bank<br/>
+                Airtel Wallet<br/><br/>
+                 payment charges: Nil<br/>
+            </td>
+            <td>
+            </td>
+            <td>
+                <img class="qr-code" src="https://donation.cmdrf.kerala.gov.in//assets/images/upi-airtel.png"/>
```
Comment (bhumilharia) [inline mainapp/templates/mainapp/contributor_form.html]: There are double slashes here (and other assets links). They don't cause any problems, but just FYI.

**Comment by bhumilharia** [comment]
Change works the same as previously. I haven't checked the account numbers and QR this time, assuming that didn't change. Good to go.

File: mainapp/templates/mainapp/contributor_form.html (hunk 3 above)
Comment (aravindharinair) [reply mainapp/templates/mainapp/contributor_form.html]: Oh. Ok. My bad. Should I rectify that now? Or is it fine if I correct it the next time I edit something?

**Comment by aravindharinair** [comment]
On a side, note, how do you set up postgres from scratch again? Something went wrong with my testing env and now it won't `git pull` ... Tried `sudo rm -R rescuekerala` from postgres user and it's asking for sudo pass which I don't know. Any help?

**Comment by bhumilharia** [comment]
@aravindharinair Didn't follow you. What do yo want to do? Can chat on slack if you want, I'm `@bhumil` there

**Comment by DextersHub** [comment]
now the commit looks clean.

**Comment by malgamves** [comment]
Could we somehow add the request on issue #509 here so we can merge them together?

**Comment by aravindharinair** [comment]
@malgamves for that, the CMO should provide an option for that, right? Like @yeaske mentioned in #509 , it's out of scope for this repo as we're not handling payments directly. Until CMO adds it, we can't provide it here.

File: mainapp/templates/mainapp/contributor_form.html (hunk 3 above)
Comment (bhumilharia) [reply mainapp/templates/mainapp/contributor_form.html]: Later is fine

**Comment by vrthra** [comment]
Marking this as tested.

**Comment by riginoommen** [comment]
@aravindharinair Can you please rebase this branch

File: mainapp/templates/mainapp/contributor_form.html (hunk 3 above)
Comment (jebu) [reply mainapp/templates/mainapp/contributor_form.html]: Would be better to host the images locally, adds unnecessary dependency on the CMDRF site, it was down when I was checking, causing broken images

**Comment by aravindharinair** [comment]
@riginoommen Sure.

**Comment by arjunkomath** [comment]
@aravindharinair I can upload these images to S3 and share the links now.

**Comment by arjunkomath** [comment]
Here are the links:
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/gateway-airtel.jpg
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/gateway-federalbank.jpg
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/gateway-hdfc.jpg
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/gateway-paytm.jpg
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/gateway-sbi.jpg
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/gateway-southindianbank.jpg
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/upi-airtel.png
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/upi-federalbank.png
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/upi-paytm.png
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/upi-sbi.png
https://s3-ap-southeast-1.amazonaws.com/keralarescue/images/upi-southindianbank.png

File: mainapp/templates/mainapp/contributor_form.html (hunk 3 above)
Comment (aravindharinair) [reply mainapp/templates/mainapp/contributor_form.html]: @jebu Waiting for @arjunkomath to upload images to S3

Comment (arjunkomath) [reply mainapp/templates/mainapp/contributor_form.html]: @aravindharinair Its done, I've added the links.

**Comment by arjunkomath** [comment]
@aravindharinair Please check the merge conflict

**Comment by aravindharinair** [comment]
screenshot from 2018-08-18 15-35-32

@arjunkomath , I have no idea what the conflict is...
I basically copy pasted the style.css file from the master repo and still a build error :/

**Comment by aravindharinair** [comment]
Any ideas?

**Comment by aravindharinair** [comment]
Someone do check this out. Test and verify details.

**Comment by arjunkomath** [comment]
@aravindharinair Please Squash the commits.

**Comment by aravindharinair** [comment]
@arjunkomath Unable to squash and push. It's saying there's conflict while `git push`.
Trying to figure it out.

**Comment by riginoommen** [comment]
@aravindharinair beacuse do force push or clean the origin working branch with upstream

**Comment by arjunkomath** [comment]
@riginoommen http://donation.cmdrf.kerala.gov.in/ is down since last 15 minutes, please prioritise this PR.

**Comment by riginoommen** [comment]
@arjunkomath need the rebase. again

**Comment by aravindharinair** [comment]
@riginoommen css file conflict... :/

**Comment by arjunkomath** [comment]
@aravindharinair please squash those commits and do a force push.

**Comment by aravindharinair** [comment]
@arjunkomath 
Adding css causes conflict. Gonna add them directly to HTML and try.

**Comment by aravindharinair** [comment]
Done and done. Test, verify and push...

**Comment by arjunkomath** [comment]
@aravindharinair you need to pull master once more, few PRs were merged.

**Comment by aravindharinair** [comment]
@arjunkomath Ok. On it.

**Comment by jilvin** [comment]
@aravindharinair Can you guys hold on merge a bit. I was making some good changes on your fork.

**Comment by aravindharinair** [comment]
@jilvin yeah, sure. What's up?

**Comment by arjunkomath** [comment]
Wait for @jilvin changes.

**Comment by aravindharinair** [comment]
Solved js conflict. Waiting for @jilvin

**Comment by riginoommen** [comment]
@aravindharinair Again you need to squash 20 =>1 commit

**Comment by aravindharinair** [comment]
@riginoommen will do once @jilvin pushes his pr...

**Comment by aravindharinair** [comment]
@riginoommen in the meantime, how can I do that?
`git log` only shows 8 commits. :/

**Comment by riginoommen** [comment]
@aravindharinair As i shown you before https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

8 commits only???. so what i think is your Working branch is messed up now.

**Comment by riginoommen** [comment]
@aravindharinair So please clone this`aravindharinair:master` branch and retry the squash operation

**Comment by aravindharinair** [comment]
@riginoommen It looks pretty messed up :/
Will try cloning and squash...

**Comment by aravindharinair** [comment]
@riginoommen , nope. no use.
`git log` returns only 12

**Comment by riginoommen** [comment]
@aravindharinair clone a new branch with upstream and move your commits or do cherry picking
https://git-scm.com/docs/git-cherry-pick

**Comment by aravindharinair** [comment]
Um. @riginoommen could you explain what you meant?
Should I refork again and work from there?

**Comment by jilvin** [comment]
@aravindharinair I have made the changes and sent a pr to your fork. Can you check?

**Comment by aravindharinair** [comment]
Doesn't the maintainers have an option to 'Squash and Merge' while merging to master?

**Comment by aravindharinair** [comment]
@jilvin Got the PR but there are conflicts...

**Comment by riginoommen** [comment]
@aravindharinair  
1) i will say merge @jilvin changes to your fork
2) then create a new branch from upstream 'git fetch upstream master:fix/payment'
3) switch to current working branch 
4) Do cherry picking or reset the  branch to move changes 
5) Verify changes
5) commit once when you feel ready

**Comment by aravindharinair** [comment]
@riginoommen now it's showing 4 :/

screenshot from 2018-08-18 17-40-36

I seriously don't know what's happening here...

**Comment by aravindharinair** [comment]
Currently, I think only the mess of commits is there. The site is working as expected. If the maintainers have an option to 'Squash and Merge', that would clean it up.
Can someone else also test the site?
I'll see how I can clean it up...

**Comment by jilvin** [comment]
Fixes #90, #52, #36.

**Comment by jilvin** [comment]
@vrthra Can you test this again and approve the pr?

**Comment by aravindharinair** [comment]
@riginoommen Deleted my copy of repo, reforked and built again. :D
Single commit now.

Closing this. Move discussions to #614

**Pull request closed without merging**